### PR TITLE
Add publish workflow

### LINKED ISSUE #8: Add publish GitHub Action workflow
Automagically publishing on each GitHub Release is nice! 😎 It means you don't have to manually do `git clone`, `npm install`, and `npm publish` after you write up your release notes.

### PR DESCRIPTION
**Pull request by jcbhmr**
🛑 You'll need to add an $NPM_TOKEN secret to the repo!

closes #8

### CODE REVIEW

**Comment by jimmywarting** [comment]
TY, added NPM_TOKEN to secrets, hope this one work!

How dose it handle minor, major and breaking PRs?
And i would like for things to generate a Github Release as well if possible and generate the changes that have happened over time. 

I have used semantic-release before and it have worked out grate for me before. 

Just have to create a good PR title that is prefixed with things:

https://semantic-release.gitbook.io/semantic-release/

image

**Comment by jcbhmr** [comment]
> How dose it handle minor, major and breaking PRs?
> And i would like for things to generate a Github Release as well if possible and generate the changes that have happened over time.

This particular workflow makes it so that you can create a GitHub Release https://github.com/jimmywarting/whatwg-worker/releases and then when you do it will run `npm publish` against whatever content is in the repo at that tag/release you just made. You still have to **manually do version bumping**, but this sort of workflow could tie in well with another workflow in the future that just creates a GitHub Release  which would  trigger this workflow to run and `npm publish`.

I've found that this basic workflow combined with the GitHub auto release notes is a pretty good low-effort baseline  for most npm projects. If the project gets big and  you can't keep track of major/minor/patch stuff, then yeah a auto semantic release thing is useful! 👍 The advantage of  keeping  small and "fit to size" is that you don't  _have to conform  to PR or commit conventions_ unless your project  becomes big and you add that later. That's my preference  for  project management though, you  do you 😊

image

You can further customize those auto-generated release  notes to create a changelog in the https://github.com/jimmywarting/whatwg-worker/releases tab.
https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

Here's a good example of  another projkect where I think this sort of GitHub Releases thing was used well: https://github.com/Andrew-Chen-Wang/github-wiki-action/releases

**Comment by jimmywarting** [comment]
Oh, did know github had things like auto generate releases (by using labels). Cool! must have missed it before.

i only knew it had a auto generate release not when i was creating tags more manually. cuz i have seen that "generate release notes" before also

This makes me wanna try it out without using semantic-release and by using GH labels instead. 
Thanks for sharing

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  publish-npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          cache: npm
+          registry-url: https://registry.npmjs.org/
+      - run: npm ci
```
Comment (jimmywarting) [inline .github/workflows/publish.yml:19]: again. still don't have any package lock file. so it complained again on the workflow action
using `npm install` would solve it.

Comment (jimmywarting) [reply .github/workflows/publish.yml:19]: i have changed it

**Pull request merged**